share/completions/conda.fish

```diff
@@ -0,0 +1,302 @@
+# Fish shell completions for conda, the package management system
+# https://conda.io/docs
+# Based on `conda help` as per version 4.4.11
+# Advanced completions:
+# - completion of already defined keys for `conda config`
+# - completion of already created environment where possible
+#
+# Note: conda ships with a completion for fish, automatically generated
+# from `conda -h` but it is far cruder than the completions in this file,
+# and it is by nature far too tricky to tweak to get the desired richness.
+# Hence this effort.
+
+# Filter for `complete -n` selecting a command line
+# with the given conda subcommand
+function __fish_conda_needs_command -a cmd
+    # We need to filter out -x and --xx options so as to complete
+    # the arguments of said options
+    test (string match -r '^\w+' -- (commandline -opc))[-1] = $cmd
+end
+
+# Complete using -n to select the given conda subcommand
+# and passing the rest of the arguments to `complete`
+# The goal here is to reduce clutter in the definitions below
+# This function will be deleted at the end
```

Comment:
That's probably not fantastic since the name is so generic.

If the user has a function with that name, you'll first redefine and then delete it.

Reply:
Call it `__conda` or something.

Reply:
Addressed in c2cb62709aac6e9a14465f03cf5df0b00d451772

Reply:
Addressed by 4b45243